include/object/tcb.h

```diff
@@ -44,6 +44,52 @@ static inline unsigned int setMR(tcb_t *receiver, word_t *receiveIPCBuffer,
 void tcbSchedEnqueue(tcb_t *tcb);
 void tcbSchedAppend(tcb_t *tcb);
 void tcbSchedDequeue(tcb_t *tcb);
+tcb_queue_t tcb_queue_remove(tcb_queue_t queue, tcb_t *tcb);
+
+static inline tcb_queue_t tcb_queue_prepend(tcb_queue_t queue, tcb_t *tcb)
+{
+    if (!queue.head) { /* Empty list */
+        queue.end = tcb;
+    } else {
+        tcb->tcbSchedNext = queue.head;
+        queue.head->tcbSchedPrev = tcb;
+    }
+
+    queue.head = tcb;
+
+    return queue;
```

Comment:
The `queue` here is passed and returned by value, not by pointer, so this is a new queue. This pattern is part of the point of the exercise, because it reduces global effects and lets us reason locally.

Reply:
Oops, my brain made it a pointer. If that makes verification easier than passing a pointer it make sense, but maybe add a comment mentioning why it's done.